src/components/Map/MapView.tsx

```diff
@@ -32,6 +45,30 @@ export const MapView = () => {
   const activeNodeId = useSelector(selectActiveNodeId());
   const { edgesFeatureCollection, viewState } = useSelector(selectMapState());
 
+  const waypoints = useSelector(selectAllWaypoints());
+  const isWaypointEdit = useSelector(selectIsWaypointEdit());
+  const newWaypoint = useSelector(selectIsNewWaypoint());
+
+  const handleClick = (e: MapLayerMouseEvent) => {
+    // Can only create new waypoint if the state is toggled
+    if (newWaypoint) {
+      const createdWaypoint: Waypoint = {
+        id: 0,
+        latitudeI: Math.round(e.lngLat.lat * 1e7), // Location clicked
+        longitudeI: Math.round(e.lngLat.lng * 1e7),
+        expire: Math.round(moment().add(1, "years").valueOf() / 1000), // Expires one year from today
+        lockedTo: 0, // Locked to me
+        name: "New Waypoint",
+        description: "",
+        icon: 128529, // (-_-) emoji; placeholder
```

Comment:
I would set this as 0 by default

Reply:
Set to 128205, round pushpin emoji (sually used as default emoji for maps) - not sure how Apple/Android/Webclient handles 0, but it should be able to handle this.

Reply:
I'm not concerned about compatibility necessarily since it's easy for the client to ignore it, moreso the UX of having a default emoji when the user doesn't explicitly set one